### PR TITLE
enh(ui) : tooltip in folded menu hover

### PR DESCRIPTION
**Pull request by EdytaSki**
<h1> Pull Request Template </h1>

<h2> Description </h2>

I added a tooltip with menu first level entries description.
How to test :
Hover on folded menu items (icons), the tooltip with description should appear.
When menu is opened - the tooltip dissapear.
When mouse leave the menu items the tooltip dissapear.

Fixes # (issue): None

<h2> Type of change </h2>

- [x] New functionality (non-breaking change)

<h2> Target serie </h2>

- [x] 18.10.x

<h2> How this pull request can be tested ? </h2>

I added a tooltip with menu first level entries description.
How to test :
Hover on folded menu items (icons), the tooltip with description should appear.
When menu is opened - the tooltip dissapear.
When mouse leave the menu items the tooltip dissapear.

<h2> Checklist </h2>

<h5> Community contributors & Centreon team </h5>

- [x] I followed the **coding style guidelines** provided by Centreon

<h5> Centreon team only </h5>

- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/front_src/src/components/tooltip/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import {connect} from "react-redux";
+
+const Tooltip = ({x, y, label, toggled}) => (
+    <div className={`tooltip ${toggled ? ' ' : 'hidden'}`}
```
Comment (sc979) [inline www/front_src/src/components/tooltip/index.js]: 4 spaces indentation instead of 2

File: www/front_src/src/components/tooltip/index.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        style={
+            {
+                top:y,
+                left:x
+            }
+        }
+    >{label}</div>
+)
+
+const mapStateToProps = ({tooltip}) => (
```
Comment (sc979) [inline www/front_src/src/components/tooltip/index.js]: please document the use of mapStateToProps

File: www/front_src/src/redux/reducers/tooltipReducer.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as actions from '../actions/tooltipActions';
+
+const initialState = {
+    toggled: false,
```
Comment (sc979) [inline www/front_src/src/redux/reducers/tooltipReducer.js]: 4 spaces indentation isntead of 2

File: www/front_src/src/redux/reducers/tooltipReducer.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    label: "",
+    x: 0,
+    y: 0
+}
+
+const tooltipReducer = (state = initialState, action) => {
+    if(action.type === actions.UPDATE_TOOLTIP){
+        const {data} = action;
+        return {...state, ...data};
+    }else {
```
Comment (sc979) [inline www/front_src/src/redux/reducers/tooltipReducer.js]:
```suggestion
    } else {
```

File: www/front_src/src/redux/reducers/tooltipReducer.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+const initialState = {
+    toggled: false,
+    label: "",
+    x: 0,
+    y: 0
+}
+
+const tooltipReducer = (state = initialState, action) => {
+    if(action.type === actions.UPDATE_TOOLTIP){
```
Comment (sc979) [inline www/front_src/src/redux/reducers/tooltipReducer.js]:
```suggestion
    if (action.type === actions.UPDATE_TOOLTIP) {
```

File: www/front_src/src/styles/partials/_side-nav.scss [hunk 6]
```diff
@@ -437,11 +437,7 @@ a {
       .collapsed-level-item {
         background-color: $white-color;
         .collapsed-level-item-link {
-          padding: 8px 10px 8px 47px;
-          // &:hover {
-          //   color: $white-color;
-          //   background-color: #0a78a5;
-          // }
+          padding: 8px 10px 8px 56px;
```
Comment (sc979) [inline www/front_src/src/styles/partials/_side-nav.scss]: modification from another PR not yet validated by the QA -> to remove from this PR

File: www/front_src/src/styles/partials/_side-nav.scss [hunk 7]
```diff
@@ -395,7 +395,7 @@ a {
       font-size: 0.8rem;
       background-color: $white-color;
       display: block;
-      padding: 3px 10px 3px 47px;
+      padding: 3px 10px 3px 56px;
```
Comment (sc979) [inline www/front_src/src/styles/partials/_side-nav.scss]: modification from another PR not yet validated by the QA -> to remove from this PR

File: www/front_src/src/styles/partials/_side-nav.scss [hunk 8]
```diff
@@ -13,6 +13,7 @@
   text-align: center;
   background: #e0e0e0;
   transition: all 0.2s;
+  z-index: 2;
```
Comment (sc979) [inline www/front_src/src/styles/partials/_side-nav.scss]: modification from another PR not yet validated by the QA -> to remove from this PR

Comment (EdytaSki) [reply www/front_src/src/styles/partials/_side-nav.scss]: this one is necessary for this feature too

**Pull request closed without merging**